cluster/etcd_service_discovery.go

```diff
@@ -187,7 +188,9 @@ func (sd *etcdServiceDiscovery) renewLease() error {
 
 func (sd *etcdServiceDiscovery) grantLease() error {
 	// grab lease
-	l, err := sd.cli.Grant(context.TODO(), int64(sd.heartbeatTTL.Seconds()))
+	ctx, cancel := context.WithTimeout(context.Background(), sd.heartbeatTTL)
```

Comment:
create a new option for acquire lease timeout.

Reply:
Isn't the heartbeatTTL a good value for the timeout?

Reply:
I believe it's not. The heartbeatTTL is something the game has control over and can increase or decrease based on how permissive they want to be with unstable connections. This one on the other hand is something internal that we can come up with a sweet spot value

Reply:
Well, shouldn't that sweet spot be about the same as the ETCD dial timeout? Maybe we can use the same since ultimately it's for a similar use: the deadline of a GRPC call to ETCD server

Reply:
makes sense, we can go with it then

Reply:
done